cmd/minishift/cmd/profile/profile_test.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"testing"
+
+	"github.com/minishift/minishift/pkg/testing/cli"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+)
+
+func Test_validate_args_exit_when_profilename_empty(t *testing.T) {
```

Comment:
+1 Why don't you add tests for the other two cases as a good measure?

Reply:
Or you can do a table based test again

Reply:
> Or you can do a table based test again

I also thought of it, so i tried it but for some reason the tests always pass when it should failed, i reckon because the function exits the loop never moves foreward.. is that right?

Reply:
what did you try. It might help to put it back, so that we can have a look. Or just make it three separate tests. I'm fine with that.

Reply:
I lost the code that i wrote but here's roughly what i did, i tried what we did for the iso url test, making a struct like this:
```
type testData struct {
        in string
        out string
        exitCode int
}

then looped over it calling validArgs and registering a exit handler, then matched the actual output with `testData.out`.